### ISSUE TITLE
Deprecated 'displaymath' node problem?

### ISSUE THREAD

**Issue by jakobandersen** [sphinx-doc/sphinx#5017]
Subject: Keeping the ``addnodes.displaymath`` node causes problems in visitors.

While trying to make Breathe work with the master branch I got an "unknown node" error from the ``HTMLTranslator`` (https://github.com/sphinx-doc/sphinx/blob/master/sphinx/writers/html.py#L883) with a ``displaymath`` node. The node in question is ``addnode.displaymath`` (https://github.com/sphinx-doc/sphinx/blob/master/sphinx/addnodes.py#L233).
The first "bad" commit is e675ad2ec, where it seems that the node was kept, but not the visitor methods.

Unless it's easy to add deprecated visitor methods I suggest simply removing the node type.

**Comment by tk0miya** [comment]
I'd like to reproduce the error on my local. Could you share any example for this?

**Comment by jakobandersen** [comment]
Sure, I think this minimal project captures what Breathe is doing in this case: [sphinx-displaymath.zip](https://github.com/sphinx-doc/sphinx/files/2045989/sphinx-displaymath.zip)
Changing line 6 in ``source/extensions/trigger_problem/__init__.py`` from
```py
node = sphinx.addnodes.displaymath()
```
to
```py
node = sphinx.addnodes.math_block()
```
makes it not crash.
If I understand correctly then the visitors does lookup by name, so class derivation from ``math_block`` is not working. Would replacing the ``displaymath`` class with ``displaymath = math_block`` in ``addnodes`` be ok? (that also makes the example and Breathe compile).

**Comment by tk0miya** [comment]
I merged #5025 to master branch now. Could you try it again please?

**Comment by jakobandersen** [comment]
Thanks, it works.
